File: ios/Plugin/SecurePasswordPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SecurePasswordPlugin)
public class SecurePasswordPlugin: CAPPlugin {
    private let implementation = SecurePassword()

    @objc func save(_ call: CAPPluginCall) {
        guard let data = call.getString("data"), let key = call.getString("key") else {
            call.reject("Missing parameters")
            return
        }

        do {
            try implementation.save(data: data, key: key)
            call.resolve()
        } catch {
            call.reject("Failed to save data: \(error.localizedDescription)")
        }
    }
    
    @objc func read(_ call: CAPPluginCall) {
        guard let key = call.getString("key") else {
            call.reject("Missing key")
            return
        }

        do {
            let data = try implementation.read(key: key)
            call.resolve(["value": data])
        } catch {
            call.reject("Failed to read data: \(error.localizedDescription)")
        }
    }
}

File: android/src/main/java/com/capacitor/plugins/securepsw/SecurePasswordPlugin.java
package com.capacitor.plugins.securepsw;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.JSObject;

@CapacitorPlugin(name = "SecurePassword")
public class SecurePasswordPlugin extends Plugin {

    private SecurePassword securePassword;

    @Override
    public void load() {
        securePassword = new SecurePassword(getContext());
    }

    @PluginMethod
    public void save(PluginCall call) {
        String key = call.getString("key");
        String data = call.getString("data");

        if (key == null || data == null) {
        call.reject("Please provide a key and data.");
        return;
        }

        try {
        securePassword.savePassword(key, data);
        JSObject ret = new JSObject();
        ret.put("success", true);
        call.resolve(ret);
        } catch (Exception e) {
        call.reject("Error saving password: " + e.getLocalizedMessage(), e);
        }
    }

    @PluginMethod
    public void read(PluginCall call) {
        String key = call.getString("key");

        if (key == null) {
        call.reject("Please provide a key.");
        return;
        }

        try {
        String data = securePassword.getPassword(key);
        if (data != null) {
            JSObject ret = new JSObject();
            ret.put("value", data);
            call.resolve(ret);
        } else {
            call.reject("No data found for the given key");
        }
        } catch (Exception e) {
        call.reject("Error reading password: " + e.getLocalizedMessage(), e);
        }
    }
}
